libs/application/template-api-modules/src/lib/modules/templates/p-sign-submission/p-sign-submission.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { TemplateApiModuleActionProps } from '../../../types'
+import {
+  SyslumennService,
+  Person,
+  Attachment,
+  PersonType,
+  DataUploadResponse,
+} from '@island.is/clients/syslumenn'
+import { NationalRegistry } from './types'
+import { SharedTemplateApiService } from '../../shared'
+import { getValueViaPath } from '@island.is/application/core'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+
+interface ContentData {
+  getFileContentAsBase64: {
+    content: string
+  }
+}
+
+const QUERY = `
+query GetFileContentAsBase64($input: FileContentAsBase64Input!) {
+  getFileContentAsBase64(input: $input) {
+    content
+  }
+}
+`
+@Injectable()
+export class PSignSubmissionService {
+  constructor(
+    private readonly syslumennService: SyslumennService,
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private readonly sharedTemplateAPIService: SharedTemplateApiService,
+  ) {}
+
+  async submitApplication({ application, auth }: TemplateApiModuleActionProps) {
+    const attachment: Attachment = await this.getAttachments({
+      application,
+      auth,
+    })
+    const nationalRegistryData = application.externalData.nationalRegistry
+      ?.data as NationalRegistry
+
+    const person: Person = {
+      name: nationalRegistryData?.fullName,
+      ssn: nationalRegistryData?.nationalId,
+      phoneNumber: application.answers.phone as string,
+      email: application.answers.email as string,
+      homeAddress: nationalRegistryData?.address.streetAddress,
+      postalCode: nationalRegistryData?.address.postalCode,
+      city: nationalRegistryData?.address.city,
+      signed: true,
+      type: PersonType.Plaintiff,
+    }
+    const persons: Person[] = [person]
+
+    const extraData: { [key: string]: string } =
+      application.answers.deliveryMethod === 'sendHome'
+        ? {
+            StarfsstodID: application.answers.district as string,
+          }
+        : {}
+
+    const uploadDataName = 'pkort1.0'
+    const uploadDataId = 'pkort1.0'
+
+    const result: DataUploadResponse = await this.syslumennService
+      .uploadData(persons, attachment, extraData, uploadDataName, uploadDataId)
+      .catch((e) => {
+        return {
+          success: false,
+          errorMessage: e.message,
+        }
+      })
+
+    if (!result.success) {
+      throw new Error(`Application submission failed`)
+    }
+    return { success: result.success, id: result.caseNumber }
+  }
+
+  private async getAttachments({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps): Promise<Attachment> {
+    const attachments = getValueViaPath(
+      application.answers,
+      'attachments',
+    ) as Array<{ key: string; name: string }>
+    const hasAttachments = attachments && attachments?.length > 0
```

Comment:
Probably doesn't matter since this is used for conditional checking, but `hasAttachments` will be `undefined` if `attachments` is  `undefined`, I believe.

```suggestion
    const hasAttachments = Array.isArray(attachments)
```